File: src/main/java/com/suryakn/IssueTracker/dto/TicketResponse.java
package com.suryakn.IssueTracker.dto;

import com.suryakn.IssueTracker.entity.Priority;
import com.suryakn.IssueTracker.entity.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TicketResponse {
    private Long id;
    private String title;
    private String description;
    private Status status;
    private Priority priority;
    private List<CommentDto> comments;
    private LocalDateTime createdAt;
    private LocalDateTime modifiedAt;
    private CreatedByDto created;
    private CreatedByDto assigned;
    private ProjectDto project;

    private List<SimilarTickets> similarTickets;

}

File: issue-tracker-web/src/TicketResponse.ts
export interface TicketResponse {
  // Define the properties of the Ticket type here
  id: number;
  title: string;
  description: string;
  status: string;
  priority: string;
  comments: Comment[];
  createdAt: string;
  modifiedAt: string;
  created: User;
  assigned: User;
  project: Project;
}
export interface User {
  firstName: string;
  lastname: string;
  email: string;
}
export interface Comment {
  email: string;
  comment: string;
  created: string;
  username: string;
}

export interface Project {
  id: number;
  name: string;
}
